pkg/storagesvc/archivePruner.go

```diff
@@ -55,17 +56,23 @@ func MakeArchivePruner(logger *zap.Logger, clientGen crd.ClientGeneratorInterfac
 }
 
 // pruneArchives listens to archiveChannel for archive ids that need to be deleted
-func (pruner *ArchivePruner) pruneArchives() {
+func (pruner *ArchivePruner) pruneArchives(ctx context.Context) {
 	pruner.logger.Debug("listening to archiveChannel to prune archives")
-	for archiveID := range pruner.archiveChan {
-		pruner.logger.Info("sending delete request for archive",
-			zap.String("archive_id", archiveID))
-		if err := pruner.stowClient.removeFileByID(archiveID); err != nil {
-			// logging the error and continuing with other deletions.
-			// hopefully this archive will be deleted in the next iteration.
-			pruner.logger.Error("ignoring error while deleting archive",
-				zap.Error(err),
+	for {
+		select {
+		case archiveID := <-pruner.archiveChan:
+			pruner.logger.Info("sending delete request for archive",
 				zap.String("archive_id", archiveID))
+			if err := pruner.stowClient.removeFileByID(archiveID); err != nil {
+				// logging the error and continuing with other deletions.
+				// hopefully this archive will be deleted in the next iteration.
+				pruner.logger.Error("ignoring error while deleting archive",
+					zap.Error(err),
+					zap.String("archive_id", archiveID))
+			}
+		case <-ctx.Done():
```

Comment:
If we stop listening on the channel we should close channel here.

```
close(pruner.archiveChan)
```

Reply:
done